packages/canon/report.api.md

```diff
@@ -0,0 +1,2628 @@
+                | undefined;
+              md?:
+                | 'none'
+                | 'xs'
+                | 'sm'
+                | 'md'
+                | 'lg'
+                | 'xl'
+                | 'xxs'
+                | 'xxl'
+                | undefined;
+              lg?:
+                | 'none'
+                | 'xs'
+                | 'sm'
+                | 'md'
+                | 'lg'
+                | 'xl'
+                | 'xxs'
+                | 'xxl'
+                | undefined;
+              xl?:
+                | 'none'
+                | 'xs'
+                | 'sm'
+                | 'md'
+                | 'lg'
+                | 'xl'
+                | 'xxs'
+                | 'xxl'
+                | undefined;
+              '2xl'?:
+                | 'none'
+                | 'xs'
+                | 'sm'
+                | 'md'
+                | 'lg'
+                | 'xl'
+                | 'xxs'
+                | 'xxl'
+                | undefined;
+            }
+          | undefined
+        )
+      | ResponsiveArray<
+          2 | 1 | 3 | 4 | 5 | 6,
+          'none' | 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxs' | 'xxl' | null
+        >;
+  } & {
+    color?:
+      | 'error'
+      | 'primary'
+      | 'secondary'
+      | {
+          light?: 'error' | 'primary' | 'secondary' | undefined;
+          dark?: 'error' | 'primary' | 'secondary' | undefined;
+        }
+      | undefined;
+    background?:
+      | 'transparent'
+      | 'background'
+      | 'elevation1'
+      | 'elevation2'
+      | {
+          light?:
+            | 'transparent'
+            | 'background'
+            | 'elevation1'
+            | 'elevation2'
+            | undefined;
+          dark?:
+            | 'transparent'
+            | 'background'
+            | 'elevation1'
+            | 'elevation2'
+            | undefined;
+        }
+      | undefined;
+  },
+) => string) & {
+  properties: Set<
+    | 'alignItems'
+    | 'background'
+    | 'border'
+    | 'borderRadius'
+    | 'boxShadow'
+    | 'color'
+    | 'display'
+    | 'flexDirection'
+    | 'flexWrap'
+    | 'gap'
+    | 'justifyContent'
+    | 'margin'
+    | 'marginBottom'
+    | 'marginLeft'
+    | 'marginRight'
+    | 'marginTop'
+    | 'padding'
+    | 'paddingBottom'
+    | 'paddingLeft'
+    | 'paddingRight'
+    | 'paddingTop'
+    | 'p'
+    | 'm'
+    | 'mt'
+    | 'mr'
+    | 'mb'
+    | 'ml'
+    | 'mx'
+    | 'my'
+    | 'pt'
+    | 'pr'
+    | 'pb'
+    | 'pl'
+    | 'px'
+    | 'py'
+    | 'marginX'
+    | 'marginY'
+    | 'paddingX'
+    | 'paddingY'
+  >;
+};
+
+// @public (undocumented)
+export const Button: ({
+  size,
+  variant,
+  children,
+  disabled,
+  iconStart,
+  iconEnd,
+  ...props
+}: ButtonProps) => React_2.JSX.Element;
+
+// @public
+export interface ButtonProps {
+  // (undocumented)
+  children: React_2.ReactNode;
+  // (undocumented)
+  disabled?: boolean;
+  // (undocumented)
+  iconEnd?: IconNames;
```

Comment:
Just highlighting that icons system is gonna need work, but let's tackle that separately. Right now I'm seeing the list of icons is hardcoded, although that probably just needs a `| string`. I do also think it needs to be possible to pass in some form of SVG icon here too though. Basically we'll probably want a magic `IconElement` type or such that just does everything we need 😁. I don't think what you got here is too far off though

Reply:
You're right we can improve the API for it for sure. That was an explicit choice to list only the icons we want in the design system. We chose Remix as our library of choice and they have 1700 icons. We just can't ship everything.

Adding new icons should be possible in the IconProvider though.